python/lsst/daf/butler/datastores/chainedDatastore.py

```diff
@@ -1116,6 +1116,11 @@ def transfer_from(
             # child datastores.
             source_datastores = (source_datastore,)
 
+        refs = list(refs)
```

Comment:
This seems wrong to me but we've declared `refs` to be `Iterable` and that doesn't support `len`.